### PR TITLE
Add dcgan example

### PR DESCRIPTION
**Pull request by ReDeiPirati**
Hi FloydHub Team,

This PR is a Sai requests: he has asked me to port the [dcgan pytorch](https://github.com/pytorch/examples/tree/master/dcgan) on FloydHub and write a simple guide.

Here's some things to do before merge this PR:

- Fork my [repo](https://github.com/ReDeiPirati/dcgan) with the FloydHub account
- Clone or Dowload/Upload these DS([LSUN](https://github.com/fyu/lsun), [CIFAR-10](https://www.floydhub.com/tarunw07/datasets/cifar-10-batches-py), [imagenet](http://www.image-net.org/), [LFW](https://www.floydhub.com/samit/datasets/lfw)) to make them authored by FloydHub
- Train the model for about 100-300 epochs(`--niter` flags) to make the checkpoints available

I will continue to test the code and fix typos.

### CODE REVIEW

**Comment by houqp** [comment]
~~Thanks for the awesome guide! I have forked your project into our official github org and granted you write access to the repo. You should be able to develop directly on that repo now :)~~

EDIT: I deleted the repo from our account since we think it's better to reference the repo from your account directly in the doc so you get full ownership and all the credits :)

**Comment by ReDeiPirati** [comment]
Thanks for the kind words, really appreciated!! 
Ok, so I will replace every `floydhub/` reference with my account reference on floydhub and we will be ready to merge this PR. Is it ok for you?

**Comment by houqp** [comment]
Yup, I think that would be better. Just make sure you don't delete the resources from your account ;)

**Comment by ReDeiPirati** [comment]
Fix typos, improved docs and replaced floydhub reference.

I think we are ready to merge!

File: docs/examples/dcgan.md
Comment (houqp) [inline docs/examples/dcgan.md]: Suggestion, can we use `is_available` function from http://pytorch.org/docs/master/cuda.html to determine if GPU is available? Then we don't need to manually change the code.

The line number link here can easily go stale if you make any change to the source file.

Comment (ReDeiPirati) [reply docs/examples/dcgan.md]: Yes, absolutely!! It was a terrible solution and I've forgotten to improve it. Thanks @houqp for the fix!

File: docs/examples/dcgan.md
Comment (houqp) [inline docs/examples/dcgan.md]: Nitpick, I think it's fine to hardcode and promote your own dataset name here :)

This makes it easier for users to try it out too since they can skip the search process and jump right into the training.

**Comment by houqp** [comment]
The rest looks good to me 👍 , I will wait for @saiprashanths to do the final call.

File: docs/examples/dcgan.md
Comment (ReDeiPirati) [reply docs/examples/dcgan.md]: I totally agree with you, this sounds more KISS like!

File: docs/examples/dcgan.md
Comment (saiprashanths) [inline docs/examples/dcgan.md]: Lets use a specific version of PyTorch (`--env pytorch-0.2`) to keep this doc up to date for the future

Comment (ReDeiPirati) [reply docs/examples/dcgan.md]: Done!

**Comment by saiprashanths** [comment]
LGTM! @mckayward This is another good example to add to our docs. Can you please take a look too and ensure it flows with the rest of the docs?

**Review by mckayward — APPROVED** [review]
Looks great! Thanks for putting this together, @ReDeiPirati. We'll get this merged and deployed today. I'll include your article in a small pool that I'm going to comb through and make minor language changes to, so that the language flows in a more uniform way across our entire documentation. But I don't want to let those edits get in the way of getting this out there. Nice work!

**Comment by ReDeiPirati** [comment]
Thanks to you all guys!! 👍

**Pull request merged**
